src/Gui/ViewProviderDocumentObject.cpp

```diff
@@ -343,7 +343,7 @@ void ViewProviderDocumentObject::attach(App::DocumentObject *pcObj)
 
     // We must collect the const char* of the strings and give it to PropertyEnumeration,
     // but we are still responsible for them, i.e. the property class must not delete the literals.
-    for (std::vector<std::string>::iterator it = aDisplayModesArray.begin(); it != aDisplayModesArray.end(); ++it) {
+    for (auto it = aDisplayModesArray.begin(); it != aDisplayModesArray.end(); ++it) {

```

Comment:
Why not ranged here?

Reply:
Given type is from instance variable not immediately obvious I should have left this one alone completely. Reversing auto.